File: apps/ios/Sources/CoreDomain/Plants/PlantPhoto.swift
import Foundation

/// A photo attached to a plant.
///
/// Source: packages/api-contracts/openapi.yaml, `PlantPhoto`.
public struct PlantPhoto: Equatable, Sendable, Identifiable {
    public let id: String
    public let plantId: String
    public let mediaId: String
    public let isPrimary: Bool
    public let createdAt: Date

    public init(id: String, plantId: String, mediaId: String, isPrimary: Bool, createdAt: Date) {
        self.id = id
        self.plantId = plantId
        self.mediaId = mediaId
        self.isPrimary = isPrimary
        self.createdAt = createdAt
    }
}

File: services/api/src/modules/plants-inventory/domain/plant-photo.ts
/**
 * A plant's attached photo. Append-only, like `media.media_record` itself —
 * no update function exists here beyond the `is_primary` flag flip, which
 * `SetPrimaryPlantPhoto` implements as clear-then-set across the whole
 * table, not a mutation of one row's own fields (see
 * `application/plant-photo-repository.ts`).
 *
 * Source: migrations/1784900000000_plants-observations-tasks-baseline.sql,
 * `plants_inventory.plant_photo`.
 */

import type { Uuid } from '../../../shared/identifiers/uuid.js';

export interface PlantPhoto {
  readonly id: Uuid;
  readonly plantId: Uuid;
  readonly mediaId: Uuid;
  readonly isPrimary: boolean;
  readonly createdAt: Date;
}

export function createPlantPhoto(
  id: Uuid,
  plantId: Uuid,
  mediaId: Uuid,
  isPrimary: boolean,
  now: Date,
): PlantPhoto {
  return { id, plantId, mediaId, isPrimary, createdAt: now };
}
